src/session_impl.cpp

```diff
@@ -5189,10 +5243,37 @@ namespace aux {
 	void session_impl::update_dht_settings()
 	{
 #ifndef TORRENT_DISABLE_DHT
-		bool const prefer_verified_nodes = m_settings.get_bool(
-			settings_pack::dht_prefer_verified_node_ids);
-
-		m_dht_settings.prefer_verified_node_ids = prefer_verified_nodes;
+#define SET_BOOL(name) \
+		m_dht_settings.name = m_settings.get_bool( \
+			settings_pack::dht_ ## name )
+#define SET_INT(name) \
+		m_dht_settings.name = m_settings.get_int( \
+			settings_pack::dht_ ## name )
+
+		SET_INT(max_peers_reply);
+		SET_INT(search_branching);
+		SET_INT(max_fail_count);
+		SET_INT(max_torrents);
+		SET_INT(max_dht_items);
+		SET_INT(max_peers);
+		SET_INT(max_torrent_search_reply);
+		SET_BOOL(restrict_routing_ips);
+		SET_BOOL(restrict_search_ips);
+		SET_BOOL(extended_routing_table);
+		SET_BOOL(aggressive_lookups);
+		SET_BOOL(privacy_lookups);
+		SET_BOOL(enforce_node_id);
+		SET_BOOL(ignore_dark_internet);
+		SET_INT(block_timeout) * 60;
+		SET_INT(block_ratelimit);
+		SET_BOOL(read_only);
+		SET_INT(item_lifetime);
+		SET_INT(upload_rate_limit);
+		SET_INT(sample_infohashes_interval);
+		SET_INT(max_infohashes_sample_count);
+		SET_BOOL(prefer_verified_node_ids);
+#undef SET_BOOL
+#undef SET_INT
```

Comment:
I wonder if it might make sense to scrap `dht::settings` altogether, and just pass in `aux::session_settings` or `settngs_interface` into the dht nodes. It would save this awkward conversion of all the DHT settings. The abstraction/separation of the DHT from the main library is already compromised, and even more so with this patch.